client/lib/restore-last-path/index.js

```diff
@@ -35,37 +48,73 @@ function validatePath( path ) {
 }
 
 function getSavedPath() {
-	return new Promise( ( resolve, reject ) => {
-		readLastPath()
-			.then( ( lastPath ) => {
-				validatePath( lastPath );
-				resolve( lastPath );
-			} )
-			.catch( ( reason ) => reject( reason ) );
-	} );
+	const lastPath = readLastPath();
+	try {
+		validatePath( lastPath );
+		return lastPath;
+	} catch ( e ) {
+		debug( e );
+	}
 }
 
-function savePath( path ) {
+/**
+ * Validate the passed path & save it to the local device for future retrieval
+ *
+ * We're intentionally using store / localStorage in this module.
+ * localForage is usually preferable, but would mean a signifcant
+ * refactor of server/bundler/loader to use an asynchronous API there.
+ *
+ * @param  {string} path The current location within calypso to be potentially saved
+ * @return {Promise}      * Rejected if path validation fails or the saved path did not change
+ *                        * Resolved if the path was saved
+ */
+export function savePath( path ) {
 	return new Promise( ( resolve, reject ) => {
+		if ( ! path || path === '/' ) {
+			store.remove( LAST_PATH );
+			return resolve( 'emptied saved path' );
+		}
+
 		try {
 			validatePath( path );
 		} catch ( e ) {
 			return reject( e );
 		}
 
-		readLastPath()
-			.then( ( lastPath ) => {
-				if ( lastPath === path ) {
-					return reject( 'path is identical' );
-				}
-				localforage.setItem( LAST_PATH, path ).then( () => resolve() );
-			} )
-			.catch( ( reason ) => reject( reason ) );
+		const lastPath = readLastPath();
+		if ( lastPath === path ) {
+			return reject( 'path is identical' );
+		}
+		store.set( LAST_PATH, path );
+		resolve( 'saved path: ' + path );
 	} );
 }
 
-export default {
-	isWhitelistedForRestoring,
-	getSavedPath,
-	savePath,
-};
+// Store a module-level variable to enforce restoreLastSession only runs once per session
+let isFirstRun = true;
```

Comment:
We expect to get a boolean from this function. What will `once` do? Return undefined?